src/raw.rs

```diff
@@ -372,59 +443,281 @@ impl Future for Connect {
+    /// let req = connected_client.delete(key);
+    /// 
+    /// let key = String::from("TiKV");
+    /// let req = connected_client.delete(&key);
+    /// 
+    /// let key = "TiKV";
+    /// let req = connected_client.delete(key);
+    /// ```
+    pub fn delete<'client, 'key: 'client>(&'client self, key: impl Into<KeyRef<'key>>) -> Delete<'client, 'key> {
+        Delete::new(self, key.into())
+    }
+
+    /// Create a new [`BatchDelete`](struct.BatchDelete.html) request.
+    /// 
+    /// Once resolved this request will result in the deletion of the given keys.
+    /// 
+    /// ```rust,no_run
+    /// use tikv_client::{Config, raw::Client};
+    /// use futures::Future;
+    /// let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// let connected_client = connecting_client.wait().unwrap();
+    /// // This is the most explicit form:
+    /// let keys = vec![&b"TiKV"[..], &b"TiDB"[..]];
+    /// let req = connected_client.batch_delete(keys);
+    /// req.wait();
+    /// 
+    /// // Other possibilities:
+    /// let keys = vec!["TiKV", "TiDB"];
+    /// let req = connected_client.batch_delete(keys);
+    /// 
+    /// let (string1, string2) = (String::from("TiKV"), String::from("TiDB"));
+    /// let keys = vec![&string1, &string2];
+    /// let req = connected_client.batch_delete(keys);
+    /// 
+    /// let (key1, string2) = (String::from("TiKV"), String::from("TiDB"));
+    /// let keys = vec![&string1, &string2];
+    /// let req = connected_client.batch_delete(keys);
+    /// ```
+    pub fn batch_delete<'client, 'keys: 'client>(&'client self, keys: impl IntoIterator<Item=impl Into<KeyRef<'keys>>>) 
+    -> BatchDelete<'client, 'keys, impl Iterator<Item=KeyRef<'keys>>> {
+        BatchDelete::new(self, keys.into_iter().map(Into::into))
+    }
+
+    /// Create a new [`Scan`](struct.Scan.html) request.
+    /// 
+    /// Once resolved this request will result in a scanner over the given keys.
+    /// 
+    /// If not passed a `limit` parameter, it will default to `u32::MAX`.
+    /// 
+    /// ```rust,no_run
+    /// use tikv_client::{KeyRef, Config, raw::Client};
+    /// use futures::Future;
+    /// let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// let connected_client = connecting_client.wait().unwrap();
+    /// // This is the most explicit form:
+    /// let inclusive_range = KeyRef::from("TiKV")..=KeyRef::from("TiDB");
+    /// let req = connected_client.scan(inclusive_range, 2);
+    /// req.wait();
+    /// 
+    /// // Other possibilities:
+    /// let exclusive_range = KeyRef::from("TiKV")..KeyRef::from("TiDB");
+    /// let req = connected_client.scan(exclusive_range, None);
+    /// ```
+    pub fn scan<'client, 'keys, Bounds>(&'client self, range: Bounds, limit: impl Into<Option<u32>>) -> Scan<'client, 'keys, Bounds> 
+    where Bounds: RangeBounds<KeyRef<'keys>> {
+        use std::u32::MAX;
```

Comment:
If the limit could be potentially large, Scan and BatchScan should definitely be Stream so it can do multiple round of scan. Otherwise the default value of MAX will blow up memory.

Reply:
Unfortunately, the scan rpc returns RepeatedField of KvPair. To make it a Stream, https://github.com/tikv/tikv/issues/3727 needs to be solved first.